.github/workflows/s3-rollout.yml

```diff
@@ -0,0 +1,90 @@
+---
+name: Rollout S3 buckets
+on:  # yamllint disable-line rule:truthy
+  workflow_call:
+    inputs:
+      # Scaleway information
+      s3-region:
+        description: Region to deploy to
+        required: true
+        type: string
+      s3-provider:
+        description: Provider to deploy to
+        required: true
+        type: string
+    secrets:
+      S3_ACCESS_KEY:
+        description: API key ID
+        required: true
+      S3_SECRET_KEY:
+        description: API key secret
+        required: true
+env:
+  # Default lifecycle.xml config
+  default-lifecycle: |
+    <LifecycleConfiguration>
+      <Rule>
+        <ID>Remove uncompleted uploads</ID>
+        <Filter>
+          <Prefix>/</Prefix>
+        </Filter>
+        <Status>Enabled</Status>
+        <AbortIncompleteMultipartUpload>
+          <DaysAfterInitiation>1</DaysAfterInitiation>
+        </AbortIncompleteMultipartUpload>
+      </Rule>
+    </LifecycleConfiguration>
+
+jobs:
+  generate-matrix:
+    name: Generate job matrices
+    runs-on: ubuntu-latest
+    outputs:
+      buckets: ${{ steps.filtered.outputs.changes }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
```

Comment:
Good idea, we may want to also use this for the k8s workflows (in a different PR)